pandas/core/groupby/grouper.py

```diff
@@ -516,6 +516,10 @@ def __init__(
                     t = self.name or str(type(self.grouper))
                     raise ValueError(f"Grouper for '{t}' not 1-dimensional")
                 self.grouper = self.index.map(self.grouper)
```

Comment:
Instead of using `map` it seems more appropriate to use `_transform_index`. Then no special handling of MultiIndex is required. The one groupby test that fails when doing this is `test_groupby_grouper_f_sanity_checked`, where the use of `map` here causes confusing errors down the line, which I _think_ is the reason for the length check on lines 523-526. By switching to `_transform_index`, the error message is instead `TypeError: 'Timestamp' object is not subscriptable` which looks to me to be the right error message.

Reply:
Thanks @rhshadrach, this solves original issue in a more consistent way.

Yet, what should be done on a failing test?

Reply:
If using `_transform_index` is correct here, then I think the test can be changed to check for a TypeError with the appropriate message to match. However, I realize now that `_transform_index` should only be used on callables - it's not clear to me from the code that `self.grouper` will always be a callable here.

Reply:
@rhshadrach, if making a change to tests that do not relate to the issue under PR does not break any rules, I'm on it.
So far `_transform_index` seems to produce desired behavior, going to take a closer look.

Reply:
Though it is not obvious from the issue being solve by this PR, the test is related to the fix - so it's okay to change. I would not remove the test, but rather make it pass by checking for the new (and sensible) error message.